src/pubsub/mod_pubsub.erl

```diff
@@ -297,24 +297,24 @@ init([ServerHost, Opts]) ->
     {_, State} = init_send_loop(ServerHost),
     {ok, State}.
 
-init_backend(Opts) ->
+init_backend(Opts, Host) ->
     TrackedDBFuns = [set_state, del_state, get_state, get_states,
                      get_states_by_lus, get_states_by_bare,
                      get_states_by_full, get_own_nodes_states,
                      get_items, get_item, set_item, del_item, del_items],
     gen_mod:start_backend_module(mod_pubsub_db, Opts, TrackedDBFuns),
     mod_pubsub_db_backend:start(),
-
-    mnesia:create_table(pubsub_last_item,
-                        [{ram_copies, [node()]},
-                         {attributes, record_info(fields, pubsub_last_item)}]).
-
+    case is_last_item_cache_enabled(Host) of
+        false -> ok;
+        DB -> pubsub_cache:create_table(DB)
+    end.
+    
 store_config_in_ets(Host, ServerHost, Opts, Plugins, NodeTree, PepMapping) ->
     Access = gen_mod:get_opt(access_createnode, Opts, fun(A) when is_atom(A) -> A end, all),
     PepOffline = gen_mod:get_opt(ignore_pep_from_offline, Opts,
                                  fun(A) when is_boolean(A) -> A end, true),
     LastItemCache = gen_mod:get_opt(last_item_cache, Opts,
-                                    fun(A) when is_boolean(A) -> A end, false),
+                                    fun(A) when A == false orelse A == rdbms orelse A == mnesia -> A end, false),
```

Comment:
`A == false orelse A == rdbms orelse A == mnesia`
to 
`A == rdbms orelse A == mnesia`